src/datatypes.ts

```diff
@@ -153,6 +156,75 @@ export function deserializeDatetime(literalString: string): Date | null {
   return date;
 }
 
+/**
+ * @internal
+ * @param value Value to serialise.
+ * @returns String representation of `value`.
+ * @see https://www.w3.org/TR/xmlschema-2/#date-lexical-representation
+ */
+export function serializeDate(value: Date): string {
+  // Although the XML Schema Date is not _exactly_ an ISO 8601 string
+  // (see https://www.w3.org/TR/xmlschema-2/#deviantformats),
+  // the deviations only affect the parsing, not the serialisation.
+  // Therefore, we can just use .toISOString():
+  return value.toISOString();
+}
+
+/**
+ * @internal
+ * @param value Value to deserialise.
+ * @returns Deserialized datetime, or null if the given value is not a valid serialised datetime.
+ * @see https://www.w3.org/TR/xmlschema-2/#date-lexical-representation
+ */
+export function deserializeDate(literalString: string): Date | null {
+  // Date in the format described at
+  // https://www.w3.org/TR/xmlschema-2/#date-lexical-representation
+  // (without constraints on the value).
+  // -? - An optional leading `-`.
+  // \d{4,}- - Four or more digits followed by a `-` representing the year. Example: "3000-".
+  // \d\d-\d\d - Two digits representing the month and two representing the day of the month,
+  //             separated by a `-`. Example: "11-03".
+
+  const dateRegEx =
+    /-?\d{4,}-\d\d-\d\d/;
+  if (!dateRegEx.test(literalString)) {
+    return null;
+  }
+  
+
+  const signedDateString = literalString;
+  // The date string can optionally be prefixed with `-`,
+  // in which case the year is negative:
+  const [yearMultiplier, dateString] =
+    signedDateString.charAt(0) === "-"
+      ? [-1, signedDateString.substring(1)]
+      : [1, signedDateString];
+  const [yearString, monthString, dayString] = dateString.split("-");
+  const utcFullYear = Number.parseInt(yearString, 10) * yearMultiplier;
+  const utcMonth = Number.parseInt(monthString, 10) - 1;
+  const utcDate = Number.parseInt(dayString, 10);
+  
+  const date = new Date(
+    Date.UTC(
+      utcFullYear,
+      utcMonth,
+      utcDate
+    )
+  );
+
+  // For the year, values from 0 to 99 map to the years 1900 to 1999. Since the serialisation
+  // always writes out the years fully, we should correct this to actually map to the years 0 to 99.
+  // See
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#Individual_date_and_time_component_values
+  if (utcFullYear >= 0 && utcFullYear < 100) {
+    // Note that we base it on the calculated year, rather than the year that was actually read.
+    // This is because the year might actually differ from the value listed in the serialisation,
+    // i.e. when moving the timezone offset to UTC pushes it into a different year:
```

Comment:
I think this comment is irrelevant for `date` - since there's no time, a timezone offset can't push it into a different year. (i.e. in a timezone of `+02:00`, a time of `23:00` could mean the actual date could differ - but if there's no time, it can't be `23:00` either :)

Reply:
In this case, can deserialize essentially ignore the timezone and just return a Date object with T00:00:00?

Reply:
Yeah I think that's the way to go. It's a bit fiddly though - I'd set it to 0:00 in the timezone set in the serialisation (or UTC, if none was set), but since JS `Date` objects only support either the local time or UTC, the time you get locally might still be somewhat different (e.g. it'd be 22:00 the previous day here).

Reply:
Actually, maybe set it to 12:00? That way I think it should always match the local and UTC date regardless of the timezone, and the time part should be ignored anyway.

(Dates in JavaScript really is a mess. There's a proposal for a replacement, [Temporal](https://tc39.es/proposal-temporal/docs/), that looks like it will be added to JavaScript somewhere in the coming few years. But that doesn't help us now...

Reply:
Yeah, I was wondering what would happen for a date with a - timezone for setting at 0:00. I'll change now.